src/transformers/tokenization_utils.py

```diff
@@ -1067,21 +1089,43 @@ def batch_encode_plus(
                 "https://github.com/huggingface/transformers/pull/2674"
             )
 
-        batch_outputs = {}
+        input_ids = []
         for ids_or_pair_ids in batch_text_or_text_pairs:
             if isinstance(ids_or_pair_ids, (list, tuple)):
                 assert len(ids_or_pair_ids) == 2
                 ids, pair_ids = ids_or_pair_ids
             else:
                 ids, pair_ids = ids_or_pair_ids, None
-            outputs = self.encode_plus(
-                ids,
-                pair_ids,
-                add_special_tokens=add_special_tokens,
+
```

Comment:
so we stopped calling encode plus?

Reply:
also cant you call get_input_ids inside the loop, since it takes a tuple?
and then append without ever splitting first and second ids?

Reply:
We stopped calling `encode_plus`, which does the following -> 

- Gets strings and converts them to ids *or* gets tokens and converts them to ids *or* gets ids and does nothing with them
- Passes the resulting ids to the `prepare_for_model` method.

`batch_encode_plus` does the same thing, just with an array instead. I changed from calling `encode_plus` to `prepare_for_model` as I need to know the sequence that, once encoded, has the maximum length in that batch as well as that length. This will be used when setting `pad_to_max_length=True`, without a `max_length`, so that it pads the smaller sequences up to the longer sequence's length (useful for creating tensors out of that batch). 

I'm not sure I understand your second comment, though. Do you mean call `get_input_ids` inside the if statement?

Reply:
Tap me, I can explain first comment.

Re padding: does https://pytorch.org/docs/stable/nn.html#torch.nn.utils.rnn.pad_sequence help? `torch.nn.utils.rnn.pad_sequence(sequences, batch_first=False, padding_value=0)`

Tap me if you want to discuss the second comment!

Reply:
Agree that we could probably use pad_sequence if it makes the code simpler (I use it in `run_language_modeling`)